### ISSUE TITLE
In native code, there is no way to determine unset `selected` vs `selected: false`

### ISSUE THREAD

**Issue by chiaramooney** [react/react-native#46988]
### Description

From the JS, the developer has the choice to specify the `accessibilityState` prop. Then, they can specify any subset of 'busy', 'expanded', 'selected', 'disabled', and 'checked'. 

In the JS `selected' can be undefined or a boolean value. However, once the data is passed to the native code, accessibilityState->selected always takes a boolean value. There is no way to determine in the native code if the user has not set the 'selected' state (i.e. the control does not have selectable behavior) or if they have set 'selected' to false (i.e. the control does have selectable behavior but it is not selected)

Could we change the 'selected' value in the accessibilityState to be an std::optional, so the 'undefined' state of `selected` could be preserved on the native side? This change is needed to implement correct accessibility behavior on Windows.

### Steps to reproduce

Set a control to have accessibilityState={{}} and accessibilityState={{selected: false}}. Both show the same data for props->accessibilityState

### React Native Version

nightly

### Affected Platforms

Other (please specify)

### Output of `npx react-native info`

```text
N/A
```

### Stacktrace or Logs

```text
N/A
```

### Reproducer

https://snack.expo.dev/5HM2iIVtfaUY5z1FeHDgm

**Comment by chiaramooney** [comment]
> ⚠️	Missing Reproducible Example
> ℹ️	We could not detect a reproducible example in your issue report. Please provide either:
> * If your bug is UI related: a [Snack](https://snack.expo.dev)
> * If your bug is build/update related: use our [Reproducer Template](https://github.com/react-native-community/reproducer-react-native/generate)

https://snack.expo.dev/5HM2iIVtfaUY5z1FeHDgm

**Comment by shubhamguptadream11** [comment]
@chiaramooney Is this issue happening only on Windows? Or is it consistent across platforms?

**Comment by dnhan1707** [comment]
Hi @chiaramooney , I'm would love to tackle this issue. 
I already took a look at the code base and found that changing the type of ````selected``` to be ```std::optional``` with default is ```std::nullopt```  should be an easy fix. 
However, what come to my mind is having another attribute to it called ````selectable``` with boolean type, but that will be more work to do. Let me know what do you think?

**Comment by chiaramooney** [comment]
Great thank you!

The first option you presented of moving the type to be std::optional would be great. This way `selected` follows the same behavior convention as `expanded`.

**Comment by joevilches** [comment]
Hey @chiaramooney, looking into tackling some a11y problems this quarter and this issue came up on my radar. Just curious so I can better understand but when you say 

> This change is needed to implement correct accessibility behavior on Windows

can you give me an example of correct accessibility behavior? Like e.g. a screen reader reading something incorrect
